canvas_modules/common-canvas/src/tooltip/tooltip.jsx

```diff
@@ -107,7 +107,7 @@ class ToolTip extends React.Component {
 		const canDisplayFullText = this.canDisplayFullText(this.triggerRef);
 		const showToolTip = (
 			// show tooltip if not disabled and showToolTipIfTruncated is false
-			(!this.props.disable && !this.props.showToolTipIfTruncated) ||
+			(!this.props.disable && !this.props.showToolTipIfTruncated && !canDisplayFullText) ||
```

Comment:
This is the same as line 112. This doesn't seem right.  Why was this changed?

Reply:
I added  it to have the expected label's behavior the same as textfiled (otherwise tooltip will show in ALL header values )
I'll  update the function below then `canDisplayFullText` , see  what needs to be changed to handle both (label & textfiled)

Reply:
For the table header it might be caused by using `Tooltip` instead of TruncatedContentTooltip

Reply:
I tried it in first but i got all textfield gone from the UI
<img width="669" alt="image" src="https://user-images.githubusercontent.com/89609176/224815974-22dcc19d-bf65-4d4b-b557-823fae6f0eed.png">

Reply:
What i was trying to say in here,  is im using `TruncatedContentTooltip` not `Tooltip` (if this was your comment) and yet labels didn't work 
<img width="669" alt="image" src="https://user-images.githubusercontent.com/89609176/224852226-5ed2e3f3-93cf-4de3-a741-01c98e7207d8.png">

Reply:
You should update your PR with the correct code then figure out why it doesn't work correctly.